### PR TITLE
ansible: New playbook to force reinstall vagrant on CentOS8

### PR DESCRIPTION
**Pull request by djgalloway**
This is insane but it worked

Signed-off-by: David Galloway <dgallowa@redhat.com>

### CODE REVIEW

File: ansible/examples/centos8-vagrant.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+- hosts: all
+  become: true
+  user: ubuntu
+  vars:
+    jenkins_user: jenkins-build
+
+  tasks:
+    - name: Get the latest vagrant version
+      shell: curl -s https://releases.hashicorp.com/vagrant/ | html2text | grep '*' | grep -v '../' | head -n 1 | cut -d '_' -f2
```
Comment (guits) [inline ansible/examples/centos8-vagrant.yml:17]:
```bash
$ curl -s -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/hashicorp/vagrant/tags | jq '.[0]["name"]'
"v2.2.10"
$
```

not sure whether this is 100% sure the first element is always the latest tag though, to be double checked...

File: ansible/examples/centos8-vagrant.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    - name: Remove vagrant
+      dnf:
+        name: vagrant
+        state: absent
+
+    # https://github.com/vagrant-libvirt/vagrant-libvirt/issues/1127#issuecomment-713651332
+    - name: Library hackery
+      shell: "{{ item }}"
+      with_items:
+        - sudo dnf -y install gcc libguestfs-tools-c libvirt libvirt-devel libxml2-devel libxslt-devel make ruby-devel
```
Comment (guits) [inline ansible/examples/centos8-vagrant.yml:46]: why not using a `package` task instead?

File: ansible/examples/centos8-vagrant.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        - "sudo dnf -y install {{ latest_vagrant_url }}"
+        - cd /tmp/; wget http://vault.centos.org/8.2.2004/BaseOS/Source/SPackages/krb5-1.17-18.el8.src.rpm; rpm2cpio krb5-1.17-18.el8.src.rpm | cpio -imdV; tar xf krb5-1.17.tar.gz; cd krb5-1.17/src; LDFLAGS='-L/opt/vagrant/embedded/' ./configure; make; sudo cp lib/libk5crypto.so.3 /opt/vagrant/embedded/lib64/
```
Comment (guits) [inline ansible/examples/centos8-vagrant.yml:48]: I would probably use a `shell` task like the following for better readability:

```yaml
- name: Library hackery
  shell: |
    cd /tmp
    wget http://vault.centos.org/8.2.2004/BaseOS/Source/SPackages/krb5-1.17-18.el8.src.rpm
    rpm2cpio krb5-1.17-18.el8.src.rpm | cpio -imdV
    tar xf krb5-1.17.tar.gz; cd krb5-1.17/src
    LDFLAGS='-L/opt/vagrant/embedded/' ./configure
    make
    sudo cp lib/libk5crypto.so.3 /opt/vagrant/embedded/lib64/
```

File: ansible/examples/centos8-vagrant.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    ## Wipe out vagrant stuff
+    # From https://github.com/vagrant-libvirt/vagrant-libvirt/issues/943#issuecomment-463678158
+    - name: Wipe out vagrant stuff
+      shell: "{{ item }}"
+      become_user: "{{ jenkins_user }}"
+      with_items:
+        - rvm implode
+        - gem uninstall --all
+        - "rm -rvf /home/{{ jenkins_user }}/.vagrant.d/gems"
+        - "rm -rvf /home/{{ jenkins_user }}/.vagrant.d/plugins.json"
```
Comment (guits) [inline ansible/examples/centos8-vagrant.yml:34]: these two lines could be done in a `file` module task I guess

File: ansible/examples/centos8-vagrant.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        latest_vagrant_url: "https://releases.hashicorp.com/vagrant/{{ latest_vagrant_version.stdout }}/vagrant_{{ latest_vagrant_version.stdout }}_x86_64.rpm"
+
+    ## Wipe out vagrant stuff
+    # From https://github.com/vagrant-libvirt/vagrant-libvirt/issues/943#issuecomment-463678158
+    - name: Wipe out vagrant stuff
+      shell: "{{ item }}"
+      become_user: "{{ jenkins_user }}"
+      with_items:
+        - rvm implode
+        - gem uninstall --all
```
Comment (guits) [inline ansible/examples/centos8-vagrant.yml:32]: The same thing as for the `Library hackery` task later, I would use `shell: |`

File: ansible/examples/centos8-vagrant.yml (hunk 1 above)
Comment (djgalloway) [reply ansible/examples/centos8-vagrant.yml:17]: There's no guarantee the repo will be available at `releases.hashicorp.com/vagrant` though.

Comment (guits) [reply ansible/examples/centos8-vagrant.yml:17]: yeh that's kind of detail i'm not 100% sure

**Comment by djgalloway** [comment]
@guits made all suggested changes

**Pull request merged**